catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -765,29 +768,83 @@ FileSystem getFileSystem(Path path, Map<String, String> config) throws IOExcepti
                 .catalogPropertiesMetadata()
                 .getOrDefault(
                     config, HadoopCatalogPropertiesMetadata.FILESYSTEM_CONNECTION_TIMEOUT_SECONDS);
+
+    ExecutorService executor =
+        Executors.newSingleThreadExecutor(
```

Comment:
The thread pool concept in Java is a coarse-grained concept and should not be repeatedly created. Can you please make it as a `static final` or field value in `HadoopCatalogOpeartons`? Do not forget increase the core size of the pool if you take the suggestion.